### PR TITLE
Allow anonymous roots and provide config classes for all quarkus. config

### PR DESCRIPTION
**Pull request by gsmet**
This is all @dmlloyd and @geoand 's work. I just reordered the commits and rebased.

Let's see what CI has to say.

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
Thanks for taking care of this @gsmet !

File: core/deployment/src/main/java/io/quarkus/deployment/TestConfig.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * TODO refactor code to actually use these values
+ */
+@ConfigRoot
+public class TestConfig {
+
+    /**
+     * Number of seconds to wait for the native image to built during testing
+     */
+    @ConfigItem(defaultValue = "300")
+    Integer nativeImageWaitTime;
```
Comment (dmlloyd) [inline core/deployment/src/main/java/io/quarkus/deployment/TestConfig.java]: Hmm we probably should have made this a duration!

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/TestConfig.java]: Certainly, but it's unused currently, I just wanted to get rid of the warnings.
We should definitely make it a duration when we actually start using the class

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/TestConfig.java]: Fixed. @gsmet I force pushed to your branch for this one

**Comment by geoand** [comment]
CI seems to be misbehaving again and not running the checks...

**Comment by gsmet** [comment]
Windows CI failure is due to the Quartz - Flyway issue.

Merging.

**Pull request merged**